src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,226 @@
+     *
+     * @param ui The ui to get the input of the user.
+     */
+    public String bye(Ui ui) {
+        return Ui.showGoodbyeMessage();
+    }
+
+    /**
+     * To print the entire Task array.
+     */
+    public String showList() {
+        String str = Ui.INDENT_SEPERATOR;
+        if (taskList.isEmpty()) {
+            str = str + "There are currently no tasks in your list";
+        } else {
+            str = str + Ui.INDENT + "Here are the tasks in your list:";
+            for (int i = 0; i < taskList.size(); i++) {
+                Task t = taskList.get(i);
+                str = str + Ui.INDENT + (i + 1) + "." + t.toString();
+            }
+        }
+        str = str + Ui.INDENT_SEPERATOR;
+        return str;
+    }
+
+    /**
+     * To mark task in the array list.
+     * Show 'Task does not exist' if the position is out of range.
+     *
+     * @param position The index of the specified task in the array list
+     *                 to be marked as completed.
+     */
+    public String mark(int position) {
+        String str = "";
+        // check for error
+        if (position + 1 > taskList.size()) {
+            str = str + Ui.INDENT_SEPERATOR + Ui.INDENT + "Task does not exist";
+        } else if (position < 0) {
+            str = str + Ui.INDENT_SEPERATOR + Ui.INDENT + "Task does not exist";
+
+        } else {
+            str = str + Ui.INDENT_SEPERATOR;
+            Task t = taskList.get(position);
+            if (t.getStatusIcon().equals(" ")) {
+                str = str + Ui.INDENT + "Nice! I've marked this task as done:";
+            }
+            t.markAsDone();
+            str = str + Ui.INDENT + "  " + t.toString();
+        }
+        str = str + Ui.INDENT_SEPERATOR;
+        return str;
+    }
+
+    /**
+     * To unmark task in the array list.
+     * Show 'Task does not exist' if the position is out of range.
+     *
+     * @param position The index of the specified task in the array list
+     *                 to be marked as incompleted.
+     */
+    public String unmark(int position) {
+        String str = "";
+        // check for error
+        if (position + 1 > taskList.size()) {
+            str = str + Ui.INDENT_SEPERATOR + Ui.INDENT + "Task does not exist";
+        } else if (position < 0) {
+            str = str + Ui.INDENT_SEPERATOR + Ui.INDENT + "Task does not exist";
+
+        } else {
+            str = str + Ui.INDENT_SEPERATOR;
+            Task t = taskList.get(position);
+            if (t.getStatusIcon().equals("X")) {
+                str = str + Ui.INDENT + "OK, I've marked this task as not done yet:";
+            }
+            t.markAsUndone();
+            str = str + Ui.INDENT + "  " + t.toString();
+        }
+        str = str + Ui.INDENT_SEPERATOR;
+        return str;
+    }
+
+    /**
+     * To add a new todo into taskList.
+     *
+     * @param t New todo to be added into taskList.
+     */
+    public String todo(Todo t) {
```

Comment:
perhaps can use verbs for methods, something like addTodo(), to improve the readability.